src/lib/seam/connect/seam-http-request.ts

```diff
@@ -0,0 +1,127 @@
+import { serializeUrlSearchParams } from '@seamapi/url-search-params-serializer'
+import type { Method } from 'axios'
+
+import type { Client } from './client.js'
+import type { SeamHttpRequestOptions } from './options.js'
+import { resolveActionAttempt } from './resolve-action-attempt.js'
+import { SeamHttpActionAttempts } from './routes/index.js'
+
+interface SeamHttpRequestParent {
+  readonly client: Client
+  readonly defaults: Required<SeamHttpRequestOptions>
+}
+
+export interface SeamHttpRequestConfig<TBody, TResponseKey> {
+  readonly path: string
+  readonly method: Method
+  readonly body?: TBody
+  readonly params?: undefined | Record<string, unknown>
+  readonly responseKey: TResponseKey
+  readonly options?: Pick<SeamHttpRequestOptions, 'waitForActionAttempt'>
+}
+
+export class SeamHttpRequest<
+  const TBody,
+  const TResponse,
+  const TResponseKey extends keyof TResponse | undefined,
+> implements
+    PromiseLike<
+      TResponseKey extends keyof TResponse ? TResponse[TResponseKey] : undefined
+    >
+{
+  readonly #parent: SeamHttpRequestParent
+  readonly #config: SeamHttpRequestConfig<TBody, TResponseKey>
+
+  constructor(
+    parent: SeamHttpRequestParent,
+    config: SeamHttpRequestConfig<TBody, TResponseKey>,
+  ) {
+    this.#parent = parent
+    this.#config = config
+  }
+
+  public get responseKey(): TResponseKey {
+    return this.#config.responseKey
+  }
+
+  public get url(): URL {
+    const { client } = this.#parent
+    const baseUrl = client.defaults.baseURL
+    if (baseUrl === undefined) {
+      throw new Error('baseUrl is required')
+    }
+
+    const params = this.#config.params
+    if (params === undefined) {
+      return new URL(this.#config.path, baseUrl)
+    }
+    return new URL(
+      `${this.#config.path}?${serializeUrlSearchParams(params)}`,
```

Comment:
Can we use 
```suggestion
      `${this.#config.path}?${this.#parent.client.defaults.paramsSerializer(params)}`,
```

Reply:
I tried, but 

```
Cannot invoke an object which is possibly 'undefined'.ts(2722)
This expression is not callable.
  Not all constituents of type 'ParamsSerializerOptions | CustomParamsSerializer' are callable.
    Type 'ParamsSerializerOptions' has no call signatures.ts(2349)
(property) paramsSerializer?: ParamsSerializerOptions | CustomParamsSerializer | undefined
```

Reply:
Hmm, how about 

```
```suggestion
      `${this.#config.path}?${this.#parent.client.defaults.paramsSerializer?.(params) ?? serializeUrlSearchParams(params)}`,
```

Reply:
Same deal unfortunately:
```
This expression is not callable.
  Not all constituents of type 'ParamsSerializerOptions | CustomParamsSerializer' are callable.
    Type 'ParamsSerializerOptions' has no call signatures.ts(2349)
```

Reply:
I'll see if we can guard this but it doesn't always work in function unions

Reply:
Ah, I feel like I ran into this issue somewhere else but I can't remember when or if I solved it. We can just use `serializeUrlSearchParams` and  file an issue then to try and respect the `paramsSerializer` if we can sort out the type problem.

Reply:
https://github.com/seamapi/javascript-http/issues/68

Reply:
I think this is fixed now - checked for typeof function